gossip/server.go

```diff
@@ -125,7 +125,10 @@ func (s *server) jitteredGossipInterval() time.Duration {
 // then begins processing connecting clients in an infinite select
 // loop via goroutine. Periodically, clients connected and awaiting
 // the next round of gossip are awoken via the conditional variable.
-func (s *server) start(rpcServer *rpc.Server) {
+func (s *server) start(rpcServer *rpc.Server, stopper *util.Stopper) {
+	stopper.AddWorker()
+	defer stopper.SetStopped()
```

Comment:
We could even encapsulate this pattern in a `Stopper.RunWorker(f func())` method.

The race is that the goroutine may not get scheduled immediately, so this may stop too soon:
```
func f(stopper *util.Stopper) {
    stopper.AddWorker()
    defer stopper.SetStopped()
}

func g() {
  stopper := util.NewStopper()
  go f(stopper)
  stopper.Stop()
}
```